### ISSUE TITLE
Overwrite the mini batch training function

### ISSUE THREAD

**Issue by HansLeonardVanBrueggemann** [keras-team/keras#9063]
hello folks, 

I'm trying to implement the paper "temporal ensembling for semi-supervised learning" (https://arxiv.org/pdf/1610.02242.pdf) 

I can't figure out how to implement the mini-batch training: at each mini-batch, the code should (1) perform a prediction for the minibatch, (2) perform a prediction with dropout on for the minibatch (3) combine cross-entropy and square loss. The resulting loss is returned for the optimizer. 

I figured out how to implement that in Theano + Lasagne, but I cant figure out how to implement in Keras. Is that even possible?
